### PR TITLE
Add Events API beta

### PR DESCRIPTION
**Pull request by alisherry**
## Description
Adding beta support for the Events API.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ X ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/events/events.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { WorkOS } from '../workos';
+import { Event } from './interfaces/event.interface';
+import { List } from '../common/interfaces/list.interface';
+import { GetEventOptions } from './interfaces';
+
+export class Events {
+  constructor(private readonly workos: WorkOS) {}
+
+  async getEvents(options: GetEventOptions): Promise<List<Event>> {
```
Comment (tribble) [inline src/events/events.ts:9]: To match our other APIs, I think this should be `listEvents`

File: src/events/interfaces/event.interface.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface Event {
+  id: string;
+  event: string;
+  created_at: string;
+  data: Record<string, unknown>;
+}
```
Comment (tribble) [inline src/events/interfaces/event.interface.ts:6]: I think we can be more specific here, like we are [with webhook types](https://github.com/workos/workos-node/blob/main/src/webhooks/interfaces/webhook.interface.ts).

File: src/events/interfaces/get-events-options.interface.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface GetEventOptions {
+  events?: string[];
```
Comment (tribble) [inline src/events/interfaces/get-events-options.interface.ts:2]: Can we be more specific here? There are a finite number of valid event types.

**Review by tribble — COMMENTED** [review]
Technically Events is a list endpoint and I think we should name our SDK method to match. I also think we can be more strict with the types.

**Comment by ospahiu** [comment]
Do we usually do tests for the node SDK?

**Comment by alisherry** [comment]
@ospahiu I don't see any outside of more generic tests for particular products, might make sense to add some more tests with options but I don't think we have any other E2E ones

File: src/events/interfaces/event.interface.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  | 'connection.deactivated'
+  | 'connection.deleted'
+  | 'dsync.activated'
+  | 'dsync.deactivated'
+  | 'dsync.deleted'
+  | 'dsync.group.created'
+  | 'dsync.group.deleted'
+  | 'dsync.group.updated'
+  | 'dsync.group.user_added'
+  | 'dsync.group.user_removed'
```
Comment (tribble) [inline src/events/interfaces/event.interface.ts:103]: 💯

**Review by tribble — APPROVED** [review]

**Pull request merged**